pkg/apiserver/rest/webservice/addon.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webservice
+
+import (
+	restfulspec "github.com/emicklei/go-restful-openapi/v2"
+	"github.com/emicklei/go-restful/v3"
+
+	apis "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+)
+
+type addonWebService struct {
+}
+
+func (c *addonWebService) GetWebService() *restful.WebService {
+	ws := new(restful.WebService)
+	ws.Path("/v1/addons").
+		Consumes(restful.MIME_XML, restful.MIME_JSON).
+		Produces(restful.MIME_JSON, restful.MIME_XML).
+		Doc("api for addon management")
+
+	tags := []string{"addon"}
+
+	ws.Route(ws.GET("/").To(noop).
+		Doc("list all addons").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.QueryParameter("query", "Fuzzy search based on name or description").DataType("string")).
+		Param(ws.QueryParameter("cluster", "Cluster-based search").DataType("string")).
+		Writes(apis.ListAddonResponse{}).Do(returns200, returns500))
+
+	ws.Route(ws.POST("/").To(noop).
+		Doc("create an addon").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Reads(apis.CreateAddonRequest{}).
+		Writes(apis.AddonMeta{}))
+
+	ws.Route(ws.DELETE("/{name}").To(noop).
+		Doc("delete an addon").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("name", "identifier of the addon").DataType("string")).
+		Writes(apis.AddonMeta{}))
+
+	ws.Route(ws.GET("/{name}").To(noop).
+		Doc("show details of an addon").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("name", "identifier of the addon").DataType("string")).
+		Writes(apis.DetailAddonResponse{}))
+
+	ws.Route(ws.GET("/{name}/status").To(noop).
```

Comment:
This API doesn't make much sense, and its semantics can be directly translated into the previous API.

Reply:
They are not equivalent.
GET /:name returns the full data of the Addon. This is for showing the full page.
GET /:name/status returns the status only. This is for polling the installation status.